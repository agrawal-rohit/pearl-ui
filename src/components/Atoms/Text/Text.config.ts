export default {
  baseStyle: {
    color: {
      light: "neutral.900",
      dark: "neutral.50",
    },
    scaleFontSize: true,
  },
  variants: {
    h1: {
      fontFamily: "heading",
      fontWeight: "bold",
      fontSize: "8xl",
      lineHeight: "10xl",
    },
    h2: {
      fontFamily: "heading",
      fontWeight: "bold",
      fontSize: "5xl",
      lineHeight: "7xl",
    },
    t1: {
      fontFamily: "heading",
      fontWeight: "semibold",
      fontSize: "2xl",
      lineHeight: "4xl",
    },
    t2: {
      fontFamily: "heading",
      fontWeight: "semibold",
      fontSize: "l",
      lineHeight: "2xl",
    },
    st1: {
      fontFamily: "heading",
      fontWeight: "semibold",
      fontSize: "m",
      lineHeight: "xl",
    },
    st2: {
      fontFamily: "heading",
      fontWeight: "semibold",
      fontSize: "s",
      lineHeight: "l",
    },
    p1: {
      fontFamily: "body",
      fontWeight: "normal",
      fontSize: "m",
      lineHeight: "xl",
    },
    p2: {
      fontFamily: "body",
      fontWeight: "normal",
      fontSize: "s",
      lineHeight: "l",
    },
    caption: {
      fontFamily: "body",
      fontWeight: "normal",
      fontSize: "xs",
      lineHeight: "m",
    },
    btn1: {
      fontFamily: "body",
      fontWeight: "medium",
      fontSize: "m",
      lineHeight: "xl",
    },
    btn2: {
      fontFamily: "body",
      fontWeight: "medium",
      fontSize: "s",
      lineHeight: "m",
    },
    btn3: {
      fontFamily: "body",
      fontWeight: "medium",
      fontSize: "xs",
      lineHeight: "s",
    },
    btn4: {
      fontFamily: "body",
      fontWeight: "medium",
      fontSize: "2xs",
      lineHeight: "xs",
    },
  },
  defaults: {
    variant: "p1",
  },
};
