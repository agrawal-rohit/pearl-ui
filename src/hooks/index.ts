export { useColorModeValue } from "./useColorModeValue";
export { useComponentConfig } from "./useComponentConfig";
export { useStyledProps } from "./useStyledProps";
export { useTheme } from "./useTheme";
