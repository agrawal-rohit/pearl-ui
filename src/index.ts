export { default as Box } from "./components/Box/Box";
export { default as Button } from "./components/Button/Button";
export { default as Text } from "./components/Text/Text";
