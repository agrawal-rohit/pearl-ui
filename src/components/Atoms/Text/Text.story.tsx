import React from "react";
import { storiesOf } from "@storybook/react-native";
import Text from "./Text";
import Screen from "../Screen/Screen";
import Box from "../Box/Box";

storiesOf("Typography", module)
  .addDecorator((getStory) => <Screen>{getStory()}</Screen>)
  .add("All", () => (
    <Box flex={1}>
      <Text variant="h1">Heading 1</Text>
      <Text variant="h2">Heading 2</Text>
      <Text variant="h3">Heading 3</Text>
      <Text variant="h4">Heading 4</Text>
      <Text variant="h5">Heading 5</Text>
      <Text variant="h6">Heading 6</Text>
      <Text variant="p1">Paragraph 1</Text>
      <Text variant="p2">Paragraph 2</Text>
      <Text variant="p3">Paragraph 3</Text>
      <Text variant="p4">Paragraph 4</Text>
      <Text variant="label">Label</Text>
      <Text variant="caption">Caption</Text>
      <Box backgroundColor="primary.500">
        <Text variant="btn1">Button 1</Text>
      </Box>
      <Box backgroundColor="primary.500">
        <Text variant="btn2">Button 2</Text>
      </Box>
      <Box backgroundColor="primary.500">
        <Text variant="btn3">Button 3</Text>
      </Box>
      <Box backgroundColor="primary.500">
        <Text variant="btn4">Button 4</Text>
      </Box>
    </Box>
  ));
