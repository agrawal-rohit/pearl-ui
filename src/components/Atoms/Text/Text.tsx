import React from "react";
import { Text as RNText } from "react-native";
import responsiveSize from "../../../utils/responsiveSize";
import { useAtomicComponentConfig } from "../../../hooks/useAtomicComponentConfig";
import {
  backgroundColor,
  BackgroundColorProps,
  color,
  ColorProps,
  layout,
  LayoutProps,
  opacity,
  OpacityProps,
  spacing,
  SpacingProps,
  textShadow,
  TextShadowProps,
  typography,
  TypographyProps,
  visible,
  VisibleProps,
} from "../../../theme/src/styleFunctions";
import {
  ComponentSizes,
  ComponentVariants,
  ResponsiveValue,
  StyleFunctionContainer,
} from "../../../theme/src/types";
import { useTheme } from "../../../hooks/useTheme";

type TextStyleProps = ColorProps &
  BackgroundColorProps &
  OpacityProps &
  VisibleProps &
  LayoutProps &
  TypographyProps &
  SpacingProps &
  TextShadowProps & {
    /** The size of the text */
    size?: ResponsiveValue<ComponentSizes<"Text">>;
    /** The variant of the text */
    variant?: ResponsiveValue<ComponentVariants<"Text">>;
    /** Whether to slightly scale the font size based on the screen dimensions */
    scaleFontSize?: boolean;
  };

export const textStyleFunctions = [
  color,
  backgroundColor,
  opacity,
  visible,
  layout,
  typography,
  spacing,
  textShadow,
] as StyleFunctionContainer[];

type RNTextProps = React.ComponentProps<typeof RNText>;

export type TextProps = TextStyleProps &
  Omit<RNTextProps, keyof TextStyleProps>;

export const buildFontConfig = (
  textStyle: any,
  allowFontScaling: boolean
): object => {
  const fontWeight = textStyle.fontWeight;
  const fontStyle = textStyle.fontStyle || "normal";

  let fontSize = textStyle.fontSize;
  let lineHeight = textStyle.lineHeight;

  if (allowFontScaling) {
    fontSize = responsiveSize(fontSize);
    lineHeight = responsiveSize(lineHeight);
  }

  const initialFontFamily = textStyle.fontFamily;
  const { theme } = useTheme();

  const finalFontFamily =
    theme.fontConfig[initialFontFamily][fontWeight][fontStyle];

  return {
    fontSize,
    fontWeight,
    fontFamily: finalFontFamily,
    fontStyle,
  };
};

/**
 * Text is the component which controls the typography across your app. By default, it renders a <Text /> component
 */
const Text = React.forwardRef((rest: TextProps, ref: any) => {
  const props = useAtomicComponentConfig(
    "Text",
    rest,
    {
      size: rest.size,
      variant: rest.variant,
    },
    "primary",
    textStyleFunctions
  );

  const memoizedBuildFontConfig = React.useCallback(
    () => buildFontConfig(props.style, props.scaleFontSize),
    [props.style, props.scaleFontSize]
  );

  props.style = {
    includeFontPadding: false,
    ...props.style,
    ...memoizedBuildFontConfig(),
  };

  return (
    <RNText
      ref={ref}
      accessible={true}
      accessibilityRole="text"
      allowFontScaling
      {...props}
    >
      {props.children}
    </RNText>
  );
});

export default Text;
