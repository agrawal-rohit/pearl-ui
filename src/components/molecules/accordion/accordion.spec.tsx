import React from "react";
import { render } from "@testing-library/react-native";
import Accordion, {
  AccordionItem,
  AccordionButton,
  AccordionPanel,
} from "./accordion";
import { ThemeProvider } from "../../../theme/src/theme-context";
import Text from "../../atoms/text/text";

jest.useFakeTimers();

describe("Molecules/Accordion", () => {
  it("passes the snapshot test with default props", () => {
    const tree = render(
      <ThemeProvider>
        <Accordion>
          <AccordionItem>
            <AccordionButton>
              <Text>Test</Text>
            </AccordionButton>
            <AccordionPanel>
              <Text>Test</Text>
            </AccordionPanel>
          </AccordionItem>
        </Accordion>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test with multiple items", () => {
    const tree = render(
      <ThemeProvider>
        <Accordion>
          <AccordionItem>
            <AccordionButton>
              <Text>Test</Text>
            </AccordionButton>
            <AccordionPanel>
              <Text>Test</Text>
            </AccordionPanel>
          </AccordionItem>
          <AccordionItem>
            <AccordionButton>
              <Text>Test</Text>
            </AccordionButton>
            <AccordionPanel>
              <Text>Test</Text>
            </AccordionPanel>
          </AccordionItem>
        </Accordion>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test with allowMultiple prop", () => {
    const tree = render(
      <ThemeProvider>
        <Accordion allowMultiple>
          <AccordionItem>
            <AccordionButton>
              <Text>Test</Text>
            </AccordionButton>
            <AccordionPanel>
              <Text>Test</Text>
            </AccordionPanel>
          </AccordionItem>
          <AccordionItem>
            <AccordionButton>
              <Text>Test</Text>
            </AccordionButton>
            <AccordionPanel>
              <Text>Test</Text>
            </AccordionPanel>
          </AccordionItem>
        </Accordion>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test with allowToggle prop", () => {
    const tree = render(
      <ThemeProvider>
        <Accordion allowToggle>
          <AccordionItem>
            <AccordionButton>
              <Text>Test</Text>
            </AccordionButton>
            <AccordionPanel>
              <Text>Test</Text>
            </AccordionPanel>
          </AccordionItem>
        </Accordion>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
