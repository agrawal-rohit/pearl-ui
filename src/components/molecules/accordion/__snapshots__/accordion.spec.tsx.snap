// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Accordion passes the snapshot test with allowMultiple prop 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={{}}
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    style={
      {
        "flexDirection": "column",
        "flexWrap": "nowrap",
      }
    }
    transition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
  >
    <View
      animate={
        {
          "borderBottomWidth": 1,
          "borderColor": "#E4E9F2",
          "marginBottom": 0,
          "paddingHorizontal": 4,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
      animatedStyle={
        {
          "value": {
            "borderBottomWidth": 1,
            "borderColor": "#E4E9F2",
            "marginBottom": 0,
            "paddingHorizontal": 4,
            "paddingVertical": 12,
            "width": "100%",
          },
        }
      }
      collapsable={false}
      style={
        {
          "borderBottomWidth": 1,
          "borderColor": "#E4E9F2",
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "marginBottom": 0,
          "paddingHorizontal": 4,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityLabel="Press me!"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-start",
            "width": "100%",
          }
        }
      >
        <View
          animatedStyle={
            {
              "value": {
                "width": "100%",
              },
            }
          }
          collapsable={false}
          state={
            {
              "__state": {
                "_isReanimatedSharedValue": true,
                "_value": {
                  "width": "100%",
                },
                "addListener": [Function],
                "modify": [Function],
                "removeListener": [Function],
                "value": {
                  "width": "100%",
                },
              },
            }
          }
          style={
            {
              "width": "100%",
            }
          }
          transition={
            {
              "duration": 50,
            }
          }
        >
          <View
            animate={
              {
                "alignItems": "center",
                "justifyContent": "space-between",
              }
            }
            animatedStyle={
              {
                "value": {
                  "alignItems": "center",
                  "justifyContent": "space-between",
                },
              }
            }
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessibilityRole="text"
              accessible={true}
              allowFontScaling={true}
              animate={
                {
                  "color": "#1A2138",
                }
              }
              animatedStyle={
                {
                  "value": {
                    "color": "#1A2138",
                  },
                }
              }
              collapsable={false}
              style={
                {
                  "color": "#1A2138",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 16.099999999999998,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "includeFontPadding": false,
                  "letterSpacing": 0,
                  "lineHeight": 22,
                }
              }
            >
              Test
            </Text>
            <Text
              accessibilityLabel="chevron-down Icon"
              accessible={true}
              allowFontScaling={false}
              animate={
                {
                  "color": "#1A2138",
                }
              }
              animatedStyle={
                {
                  "value": {
                    "color": "#1A2138",
                  },
                }
              }
              collapsable={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 13.799999999999999,
                  },
                  {
                    "color": "#1A2138",
                  },
                  {
                    "fontFamily": "feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              ï„®
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      animate={
        {
          "borderBottomWidth": 1,
          "borderColor": "#E4E9F2",
          "paddingHorizontal": 4,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
      animatedStyle={
        {
          "value": {
            "borderBottomWidth": 1,
            "borderColor": "#E4E9F2",
            "paddingHorizontal": 4,
            "paddingVertical": 12,
            "width": "100%",
          },
        }
      }
      collapsable={false}
      style={
        {
          "borderBottomWidth": 1,
          "borderColor": "#E4E9F2",
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "paddingHorizontal": 4,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityLabel="Press me!"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-start",
            "width": "100%",
          }
        }
      >
        <View
          animatedStyle={
            {
              "value": {
                "width": "100%",
              },
            }
          }
          collapsable={false}
          state={
            {
              "__state": {
                "_isReanimatedSharedValue": true,
                "_value": {
                  "width": "100%",
                },
                "addListener": [Function],
                "modify": [Function],
                "removeListener": [Function],
                "value": {
                  "width": "100%",
                },
              },
            }
          }
          style={
            {
              "width": "100%",
            }
          }
          transition={
            {
              "duration": 50,
            }
          }
        >
          <View
            animate={
              {
                "alignItems": "center",
                "justifyContent": "space-between",
              }
            }
            animatedStyle={
              {
                "value": {
                  "alignItems": "center",
                  "justifyContent": "space-between",
                },
              }
            }
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessibilityRole="text"
              accessible={true}
              allowFontScaling={true}
              animate={
                {
                  "color": "#1A2138",
                }
              }
              animatedStyle={
                {
                  "value": {
                    "color": "#1A2138",
                  },
                }
              }
              collapsable={false}
              style={
                {
                  "color": "#1A2138",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 16.099999999999998,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "includeFontPadding": false,
                  "letterSpacing": 0,
                  "lineHeight": 22,
                }
              }
            >
              Test
            </Text>
            <Text
              accessibilityLabel="chevron-down Icon"
              accessible={true}
              allowFontScaling={false}
              animate={
                {
                  "color": "#1A2138",
                }
              }
              animatedStyle={
                {
                  "value": {
                    "color": "#1A2138",
                  },
                }
              }
              collapsable={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 13.799999999999999,
                  },
                  {
                    "color": "#1A2138",
                  },
                  {
                    "fontFamily": "feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              ï„®
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Molecules/Accordion passes the snapshot test with allowToggle prop 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={{}}
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    style={
      {
        "flexDirection": "column",
        "flexWrap": "nowrap",
      }
    }
    transition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
  >
    <View
      animate={
        {
          "borderBottomWidth": 1,
          "borderColor": "#E4E9F2",
          "paddingHorizontal": 4,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
      animatedStyle={
        {
          "value": {
            "borderBottomWidth": 1,
            "borderColor": "#E4E9F2",
            "paddingHorizontal": 4,
            "paddingVertical": 12,
            "width": "100%",
          },
        }
      }
      collapsable={false}
      style={
        {
          "borderBottomWidth": 1,
          "borderColor": "#E4E9F2",
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "paddingHorizontal": 4,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityLabel="Press me!"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-start",
            "width": "100%",
          }
        }
      >
        <View
          animatedStyle={
            {
              "value": {
                "width": "100%",
              },
            }
          }
          collapsable={false}
          state={
            {
              "__state": {
                "_isReanimatedSharedValue": true,
                "_value": {
                  "width": "100%",
                },
                "addListener": [Function],
                "modify": [Function],
                "removeListener": [Function],
                "value": {
                  "width": "100%",
                },
              },
            }
          }
          style={
            {
              "width": "100%",
            }
          }
          transition={
            {
              "duration": 50,
            }
          }
        >
          <View
            animate={
              {
                "alignItems": "center",
                "justifyContent": "space-between",
              }
            }
            animatedStyle={
              {
                "value": {
                  "alignItems": "center",
                  "justifyContent": "space-between",
                },
              }
            }
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessibilityRole="text"
              accessible={true}
              allowFontScaling={true}
              animate={
                {
                  "color": "#1A2138",
                }
              }
              animatedStyle={
                {
                  "value": {
                    "color": "#1A2138",
                  },
                }
              }
              collapsable={false}
              style={
                {
                  "color": "#1A2138",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 16.099999999999998,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "includeFontPadding": false,
                  "letterSpacing": 0,
                  "lineHeight": 22,
                }
              }
            >
              Test
            </Text>
            <Text
              accessibilityLabel="chevron-down Icon"
              accessible={true}
              allowFontScaling={false}
              animate={
                {
                  "color": "#1A2138",
                }
              }
              animatedStyle={
                {
                  "value": {
                    "color": "#1A2138",
                  },
                }
              }
              collapsable={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 13.799999999999999,
                  },
                  {
                    "color": "#1A2138",
                  },
                  {
                    "fontFamily": "feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              ï„®
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Molecules/Accordion passes the snapshot test with default props 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={{}}
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    style={
      {
        "flexDirection": "column",
        "flexWrap": "nowrap",
      }
    }
    transition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
  >
    <View
      animate={
        {
          "borderBottomWidth": 1,
          "borderColor": "#E4E9F2",
          "paddingHorizontal": 4,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
      animatedStyle={
        {
          "value": {
            "borderBottomWidth": 1,
            "borderColor": "#E4E9F2",
            "paddingHorizontal": 4,
            "paddingVertical": 12,
            "width": "100%",
          },
        }
      }
      collapsable={false}
      style={
        {
          "borderBottomWidth": 1,
          "borderColor": "#E4E9F2",
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "paddingHorizontal": 4,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityLabel="Press me!"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-start",
            "width": "100%",
          }
        }
      >
        <View
          animatedStyle={
            {
              "value": {
                "width": "100%",
              },
            }
          }
          collapsable={false}
          state={
            {
              "__state": {
                "_isReanimatedSharedValue": true,
                "_value": {
                  "width": "100%",
                },
                "addListener": [Function],
                "modify": [Function],
                "removeListener": [Function],
                "value": {
                  "width": "100%",
                },
              },
            }
          }
          style={
            {
              "width": "100%",
            }
          }
          transition={
            {
              "duration": 50,
            }
          }
        >
          <View
            animate={
              {
                "alignItems": "center",
                "justifyContent": "space-between",
              }
            }
            animatedStyle={
              {
                "value": {
                  "alignItems": "center",
                  "justifyContent": "space-between",
                },
              }
            }
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessibilityRole="text"
              accessible={true}
              allowFontScaling={true}
              animate={
                {
                  "color": "#1A2138",
                }
              }
              animatedStyle={
                {
                  "value": {
                    "color": "#1A2138",
                  },
                }
              }
              collapsable={false}
              style={
                {
                  "color": "#1A2138",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 16.099999999999998,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "includeFontPadding": false,
                  "letterSpacing": 0,
                  "lineHeight": 22,
                }
              }
            >
              Test
            </Text>
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Molecules/Accordion passes the snapshot test with multiple items 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={{}}
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    style={
      {
        "flexDirection": "column",
        "flexWrap": "nowrap",
      }
    }
    transition={
      {
        "dampingRatio": 1,
        "duration": 100,
        "type": "spring",
      }
    }
  >
    <View
      animate={
        {
          "borderBottomWidth": 1,
          "borderColor": "#E4E9F2",
          "marginBottom": 0,
          "paddingHorizontal": 4,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
      animatedStyle={
        {
          "value": {
            "borderBottomWidth": 1,
            "borderColor": "#E4E9F2",
            "marginBottom": 0,
            "paddingHorizontal": 4,
            "paddingVertical": 12,
            "width": "100%",
          },
        }
      }
      collapsable={false}
      style={
        {
          "borderBottomWidth": 1,
          "borderColor": "#E4E9F2",
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "marginBottom": 0,
          "paddingHorizontal": 4,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityLabel="Press me!"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-start",
            "width": "100%",
          }
        }
      >
        <View
          animatedStyle={
            {
              "value": {
                "width": "100%",
              },
            }
          }
          collapsable={false}
          state={
            {
              "__state": {
                "_isReanimatedSharedValue": true,
                "_value": {
                  "width": "100%",
                },
                "addListener": [Function],
                "modify": [Function],
                "removeListener": [Function],
                "value": {
                  "width": "100%",
                },
              },
            }
          }
          style={
            {
              "width": "100%",
            }
          }
          transition={
            {
              "duration": 50,
            }
          }
        >
          <View
            animate={
              {
                "alignItems": "center",
                "justifyContent": "space-between",
              }
            }
            animatedStyle={
              {
                "value": {
                  "alignItems": "center",
                  "justifyContent": "space-between",
                },
              }
            }
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessibilityRole="text"
              accessible={true}
              allowFontScaling={true}
              animate={
                {
                  "color": "#1A2138",
                }
              }
              animatedStyle={
                {
                  "value": {
                    "color": "#1A2138",
                  },
                }
              }
              collapsable={false}
              style={
                {
                  "color": "#1A2138",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 16.099999999999998,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "includeFontPadding": false,
                  "letterSpacing": 0,
                  "lineHeight": 22,
                }
              }
            >
              Test
            </Text>
            <Text
              accessibilityLabel="chevron-down Icon"
              accessible={true}
              allowFontScaling={false}
              animate={
                {
                  "color": "#1A2138",
                }
              }
              animatedStyle={
                {
                  "value": {
                    "color": "#1A2138",
                  },
                }
              }
              collapsable={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 13.799999999999999,
                  },
                  {
                    "color": "#1A2138",
                  },
                  {
                    "fontFamily": "feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              ï„®
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      animate={
        {
          "borderBottomWidth": 1,
          "borderColor": "#E4E9F2",
          "paddingHorizontal": 4,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
      animatedStyle={
        {
          "value": {
            "borderBottomWidth": 1,
            "borderColor": "#E4E9F2",
            "paddingHorizontal": 4,
            "paddingVertical": 12,
            "width": "100%",
          },
        }
      }
      collapsable={false}
      style={
        {
          "borderBottomWidth": 1,
          "borderColor": "#E4E9F2",
          "flexDirection": "column",
          "flexWrap": "nowrap",
          "paddingHorizontal": 4,
          "paddingVertical": 12,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityLabel="Press me!"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-start",
            "width": "100%",
          }
        }
      >
        <View
          animatedStyle={
            {
              "value": {
                "width": "100%",
              },
            }
          }
          collapsable={false}
          state={
            {
              "__state": {
                "_isReanimatedSharedValue": true,
                "_value": {
                  "width": "100%",
                },
                "addListener": [Function],
                "modify": [Function],
                "removeListener": [Function],
                "value": {
                  "width": "100%",
                },
              },
            }
          }
          style={
            {
              "width": "100%",
            }
          }
          transition={
            {
              "duration": 50,
            }
          }
        >
          <View
            animate={
              {
                "alignItems": "center",
                "justifyContent": "space-between",
              }
            }
            animatedStyle={
              {
                "value": {
                  "alignItems": "center",
                  "justifyContent": "space-between",
                },
              }
            }
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessibilityRole="text"
              accessible={true}
              allowFontScaling={true}
              animate={
                {
                  "color": "#1A2138",
                }
              }
              animatedStyle={
                {
                  "value": {
                    "color": "#1A2138",
                  },
                }
              }
              collapsable={false}
              style={
                {
                  "color": "#1A2138",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 16.099999999999998,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "includeFontPadding": false,
                  "letterSpacing": 0,
                  "lineHeight": 22,
                }
              }
            >
              Test
            </Text>
            <Text
              accessibilityLabel="chevron-down Icon"
              accessible={true}
              allowFontScaling={false}
              animate={
                {
                  "color": "#1A2138",
                }
              }
              animatedStyle={
                {
                  "value": {
                    "color": "#1A2138",
                  },
                }
              }
              collapsable={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 13.799999999999999,
                  },
                  {
                    "color": "#1A2138",
                  },
                  {
                    "fontFamily": "feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              ï„®
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
